policy-controller/src/admission.rs

```diff
@@ -209,9 +196,42 @@ impl Validate<ServerSpec> for Admission {
     }
 }
 
+impl Admission {
+    /// Detects whether two pod selectors can select the same pod
+    //
+    // TODO(ver) We can probably detect overlapping selectors more effectively. For
+    // example, if `left` selects pods with 'foo=bar' and `right` selects pods with
+    // 'foo', we should indicate the selectors overlap. It's a bit tricky to work
+    // through all of the cases though, so we'll just punt for now.
+    fn overlaps(left: &labels::Selector, right: &labels::Selector) -> bool {
+        if left.selects_all() || right.selects_all() {
+            return true;
+        }
+
+        left == right
+    }
+}
+
 #[async_trait::async_trait]
 impl Validate<ServerAuthorizationSpec> for Admission {
     async fn validate(self, _ns: &str, _name: &str, spec: ServerAuthorizationSpec) -> Result<()> {
+        if let Some(mtls) = spec.client.mesh_tls.as_ref() {
+            if spec.client.unauthenticated {
+                bail!("`unauthenticated` must be false if `mesh_tls` is specified");
+            }
+            if mtls.unauthenticated_tls {
+                let ids = mtls.identities.as_ref().map(|ids| ids.len()).unwrap_or(0);
+                let sas = mtls
+                    .service_accounts
+                    .as_ref()
+                    .map(|sas| sas.len())
+                    .unwrap_or(0);
```

Comment:
dumb code golf nit, take it or leave it: these could use `Option::map_or`
```suggestion
                let ids = mtls.identities.as_ref().map_or(0, |ids| ids.len());
                let sas = mtls
                    .service_accounts
                    .as_ref()
                    .map_or(0, |sas| sas.len());
```